Source/ARTRealtimeChannel.m

```diff
@@ -331,23 +323,79 @@ - (void)publishProtocolMessage:(ARTProtocolMessage *)pm callback:(void (^)(ARTSt
         case ARTRealtimeChannelSuspended:
         case ARTRealtimeChannelDetaching:
         case ARTRealtimeChannelDetached:
-        case ARTRealtimeChannelFailed:
-        {
+        case ARTRealtimeChannelFailed: {
             if (cb) {
-                cb(statusInvalidChannel);
+                cb(invalidChannelError);
             }
             break;
         }
-        case ARTRealtimeChannelAttached:
-        {
+        case ARTRealtimeChannelAttached: {
             if (_realtime.connection.state_nosync == ARTRealtimeConnected) {
-                [self sendMessage:pm callback:cb];
-            } else {
+                [self sendMessage:pm callback:^(ARTStatus *status) {
+                    if (cb) cb(status.errorInfo);
+                }];
+            }
+            else {
                 [self addToQueue:pm callback:queuedCallback];
+            }
+            break;
+        }
+    }
+} ART_TRY_OR_MOVE_TO_FAILED_END
+}
 
-                [self.realtime.internalEventEmitter once:[ARTEvent newWithConnectionEvent:ARTRealtimeConnectionEventConnected] callback:^(ARTConnectionStateChange *__art_nullable change) {
-                    [weakSelf sendQueuedMessages];
-                }];
+- (void)publishProtocolMessage:(ARTProtocolMessage *)pm callback:(void (^)(ARTStatus *))cb {
+ART_TRY_OR_MOVE_TO_FAILED_START(_realtime) {
+    ARTStatus *statusInvalidConnectionState = [ARTStatus state:ARTStateError info:[ARTErrorInfo createWithCode:90001 message:@"channel operation failed (invalid connection state)"]];
+    ARTStatus *statusInvalidChannelState = [ARTStatus state:ARTStateError info:[ARTErrorInfo createWithCode:90001 message:@"channel operation failed (invalid channel state)"]];
```

Comment:
Could these not be a bit more specific? (cf https://github.com/ably/ably-cocoa/issues/843)

Reply:
PR https://github.com/ably/ably-cocoa/pull/895 created.